stanza/corenlp/client.py

```diff
@@ -16,24 +17,216 @@ def __init__(self, server='http://localhost:9000'):
+    return doc
 
-    http://stanfordnlp.github.io/CoreNLP/corenlp-server.html.
+  def annotate(self, text, annotators):
+    """Return an AnnotatedDocument from the CoreNLP server.
 
-    The properties for each of the annotators can be found at
+    :param (str) text: text to be annotated
+    :param (list[str]) annotators: a list of annotator names
 
-    http://stanfordnlp.github.io/CoreNLP/annotators.html
+    See a list of valid annotator names here:
+      http://stanfordnlp.github.io/CoreNLP/annotators.html
 
-    :param text: Text to annotate.
-    :param properties: A dictionary of properties for CoreNLP.
+    :return (AnnotatedDocument): an annotated document
     """
-    properties = properties or {}
-    r = requests.get(self.server, params={'properties': str(properties)}, data=text)
-    return r.json(strict=False)
+    # TODO(kelvin): include raw text attribute for each sentence
+    doc_pb = self.annotate_proto(text, annotators)
+    return AnnotatedDocument(doc_pb)
+
+
+class Document(Sequence):
+  """A sequence of Sentences."""
+  pass
+
+
+class Sentence(Sequence):
+  """A sequence of strings."""
+  pass
+
+
+class Token(object):
+  @abstractproperty
+  def word(self):
+    pass
+
+
+class AnnotatedDocument(Document):
+  def __init__(self, doc_pb):
+    self.pb = doc_pb
+    self._sentences = [AnnotatedSentence(sent) for sent in self.pb.sentence]
+
+  def __getitem__(self, i):
+    return self._sentences[i]
+
+  def __len__(self):
+    return len(self._sentences)
+
+  @staticmethod
+  def from_dict(json_dict):
+    return AnnotatedDocument(AnnotatedDocument.dict_to_pb(json_dict))
+
+  @staticmethod
+  def dict_to_pb(json_dict):
+    sentences = [AnnotatedSentence.dict_to_pb(d) for d in json_dict['sentences']]
+    doc = CoreNLP_pb2.Document()
+    # TODO(kelvin): set doc.text and other available properties
+    doc.sentence.extend(sentences)
+    return doc
+
+  @property
+  def coref(self):
+    raise NotImplementedError
+
+  def __str__(self):
+    return self.pb.text
+
+
+# TODO(kelvin): finish specifying the Simple interface for AnnotatedSentence
+# http://stanfordnlp.github.io/CoreNLP/simple.html
+# In particular, all the methods that take arguments.
+
+
+class AnnotatedSentence(Sentence):
+  def __init__(self, sentence_pb):
+    self.pb = sentence_pb
+    self._tokens = [AnnotatedToken(tok) for tok in self.pb.token]
+
+  def __getitem__(self, i):
+    return self.pb.token[i].word
+
+  def __len__(self):
+    return len(self.pb.token)
+
+  def __str__(self):
+    return self.text
+
+  @staticmethod
+  def dict_to_pb(json_dict):
+    sent = CoreNLP_pb2.Sentence()
+    tokens = [AnnotatedToken.dict_to_pb(d) for d in json_dict['tokens']]
+    # TODO(kelvin): set other properties of sentence
+    sent.token.extend(tokens)
+    return sent
+
+  @property
+  def tokens(self):
+    return self._tokens
+
+  @property
+  def text(self):
+    return self.pb.text
+
+  @property
+  def pos_tags(self):
+    raise NotImplementedError
```

Comment:
Why do these always raise errors? Can do we a lazy check for whether the field exists and then raise if the field doesn't exist?

Also I think you want to raise and instance and not a class:

raise NotImplementedError()